preview_generator/preview/builder/archive__zip.py

```diff
@@ -114,6 +118,36 @@ def get_supported_mimetypes(cls) -> typing.List[str]:
             "application/x-tgz",
         ]
 
+    def build_jpeg_preview(
+        self,
+        file_path: str,
+        preview_name: str,
+        cache_path: str,
+        page_id: int,
+        extension: str = ".jpg",
+        size: ImgDims = None,
+        mimetype: str = "",
+    ) -> None:
+        """
+        jpeg preview is only available for .sketch file
+        """
+        if pathlib.Path(file_path).suffix != ".sketch":
```

Comment:
This broke support for file without extension. This is needed for proper support in Tracim, software which heavily use preview_generator (for technical reason, file are named without extension).

We should be able to have a file without extension and force preview_generator to handle it as a specific type. Current mecanism is based on mimetype at builder level, and on separated file extension at manager level (file_extension is used to obtain mimetype).

Reply:
Sketch files really have the `application/zip` mimetype. It should not break anything because zip files actually don't have any jpeg preview. But it could be nice to preview stretch file if file are named without extension, but at this time I don't see how to do this.

Reply:
Go on, use an unofficial mime type

Reply:
> Sketch files really have the `application/zip` mimetype. It should not break anything because zip files actually don't have any jpeg preview. But it could be nice to preview stretch file if file are named without extension, but at this time I don't see how to do this.

Just look at vtk builder for `.obj `or `.ply` file`.
`.obj` and `.ply` doesn't have real specific mimetype (in fact, these are txt file). 
Consider sketch file extension as `application/sketch` at builder level and add mimetypeMapping (see vtk builder example). This way, Preview_generator, will consider all .sketch file as "application/sketch", but will also allowing forcing file_extension at manager level and forcing mimetype at builder level.

Reply:
Thanks for your feedback, I'll have a look